openupgrade_scripts/scripts/project_todo/17.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2024 Viindoo Technology Joint Stock Company (Viindoo)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+_rename_xmlids = [
+    (
+        "note.mail_activity_data_reminder",
+        "project_todo.mail_activity_data_reminder",
+    ),
+]
+
+
+def _convert_note_tag_to_project_tags(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO project_tags (
+            color, create_uid, write_uid, name, create_date, write_date
+        )
+        SELECT color, create_uid, write_uid, name, create_date, write_date
+        FROM note_tag
+        ON CONFLICT (name) DO NOTHING;
+        """,
+    )
+
+
+def _convert_note_note_to_project_task(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE project_task ADD COLUMN note_id INTEGER;
```

Comment:
```suggestion
        ALTER TABLE project_task ADD COLUMN openupgrade_legacy_note_id INTEGER;
```
Có thể nên đổi tên cột tạm này thành cái tên j khác, ai đó có thể phát triển tính năng mà từ note ra task ở v16, và trên `project.task` họ có trường này `note_id`